core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -762,7 +768,7 @@ public synchronized void close() throws IOException {
         toClose.add(() -> stopWatch.stop().start("discovery"));
         toClose.add(injector.getInstance(Discovery.class));
         toClose.add(() -> stopWatch.stop().start("monitor"));
-        toClose.add(injector.getInstance(MonitorService.class));
+        toClose.add(nodeService.getMonitorService());
```

Comment:
this is redundant. It seems to be already closed by nodeService. It's a bit weird that NodeService closes the monitor service but doesn't call own calling the stop / start methods. Not a big deal though.